lib/ansible/module_utils/basic.py

```diff
@@ -1949,15 +1948,16 @@ def _return_formatted(self, kwargs):
                     self.warn(w)
             else:
                 self.warn(kwargs['warnings'])
+
         if self._warnings:
             kwargs['warnings'] = self._warnings
 
         if 'deprecations' in kwargs:
-            if isinstance(kwargs['deprecations'], list):
-                for d in kwargs['deprecations']:
-                    self.warn(d)
-            else:
-                self.warn(kwargs['deprecations'])
+            for d in kwargs['deprecations']:
+                if isinstance(d, SEQUENCETYPE) and len(d) == 2:
```

Comment:
Yep, when `deprecations` keys is returned directly:
* either we choose that `version`is required: value associated with `deprecations` key must be a list of pairs (message and version)
* or we choose that `version` isn't required: value can be a list of pair (message and version) and/or string (message without version)
The second choice is implemented here, let me known if you prefer the first :)